### PR TITLE
[cli] Run eslint directly rather than lint script

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Possible solution for ENG-12053

# How

Run `eslint .` instead of the script. This maybe isn't ideal because you could customize the linting command in the script before, but just opening this as some way to unblock shipping something for now.

### CODE REVIEW

File: packages/@expo/cli/src/lint/lintAsync.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const lintAsync = async (projectRoot: string) => {
 
   const manager = createForProject(projectRoot);
   try {
-    await manager.runAsync(['run', 'lint']);
+    await manager.runAsync(['run', 'eslint', '.']);
```
Comment (byCedric) [inline packages/@expo/cli/src/lint/lintAsync.ts:13]: This won't work for `npm` or `pnpm`. Here are the equivalents:
- `npm` -> `npx eslint .`
- `pnpm` -> `pnpm eslint .`

Alternatively, we could either spawn a new process or just `require('eslint/bin/eslint')` to keep it "working" for now.

Comment (brentvatne) [reply packages/@expo/cli/src/lint/lintAsync.ts]: good call! wdyt about this? https://github.com/expo/expo/pull/28332/files#diff-674041de5e2d6dc1344c60e176ff815c71ad2c76c1ac823e2c99b77022813c4fR29

File: packages/@expo/package-manager/src/node/NpmPackageManager.ts [hunk 2]
```diff
@@ -82,6 +83,11 @@ export class NpmPackageManager extends BasePackageManager {
     return this.runAsync(['uninstall', '--global', ...namesOrFlags]);
   }
 
+  runBinAsync(command: string[], options: SpawnOptions = {}) {
+    this.log?.(`> npx ${command.join(' ')}`);
+    return spawnAsync('npx', command, { ...this.options, ...options });
```
Comment (byCedric) [inline packages/@expo/package-manager/src/node/NpmPackageManager.ts:88]: Doesn't this need to be split up with `--`? E.g. `npx eslint -- --max-warnings 0`?

File: packages/@expo/package-manager/src/node/__tests__/BunPackageManager-test.ts [hunk 3]
```diff
@@ -49,6 +49,15 @@ describe('BunPackageManager', () => {
     });
   });
 
+  describe('runBinAsync', () => {
+    it('logs executed command', async () => {
+      const log = jest.fn();
+      const bun = new BunPackageManager({ cwd: projectRoot, log });
+      await bun.runBinAsync(['eslint', '.']);
+      expect(log).toHaveBeenCalledWith('> bun eslint .');
+    });
```
Comment (byCedric) [inline packages/@expo/package-manager/src/node/__tests__/BunPackageManager-test.ts:58]: Technically, you might want to test not on log, but on `spawn`. E.g.

```
const bun = new BunPackageManager({ cwd: projectRoot });
await bun.runBinAsync(['eslint', '.']);

expect(spawnAsync).toBeCalledWith(...)
```

**Review by byCedric — APPROVED** [review]

File: packages/@expo/package-manager/src/node/NpmPackageManager.ts (hunk 2 above)
Comment (brentvatne) [reply packages/@expo/package-manager/src/node/NpmPackageManager.ts:88]: seems to work fine without `--`

```
npx eslint . --help
eslint [options] file.js [file.js] [dir]

Basic configuration:
...etc
```

**Pull request merged**